test/functional/services/filter_bar.js

```diff
@@ -65,16 +78,15 @@ export function FilterBarProvider({ getService }) {
       await typeIntoReactSelect('filterOperatorList', operator);
       const params = await testSubjects.find('filterParams');
       const paramFields = await params.findAllByTagName('input');
-      await Promise.all(values.map(async (value, index) => {
-        await paramFields[index].type(value);
-        // Checks if the actual options value has an auto complete (like 'is one of' filter)
-        // In this case we need to click the active autocompletion.
-        const hasAutocompletion = await find.exists(async () => await params.findByClassName('active'));
-        if (hasAutocompletion) {
-          const activeSelection = await params.findByClassName('active');
-          await activeSelection.click();
-        }
-      }));
+      for (let i = 0; i < values.length; i++) {
+        // If we have multiple fields for the value, enter the individual
+        // array elements into each of the inputs, but if we have more array
+        // values than input fields, enter them all in the last (e.g. useful
+        // to do addFilter('extension', 'is one of', ['jpg', 'png']), which is
+        // only one input field)
+        await paramFields[Math.min(paramFields.length - 1, i)].type(values[i]);
```

Comment:
why are we falling back to paramFields.length ? will index ever be greater ? in case it will be this will use the last paramFields ? probably shouldnt ?

Reply:
Since both, Stacey and you commented already on this, I think that way of implementing it, might cause too many confusion. I will change the API slightly, so you can pass in multiple parameters to the function and they will instead be inserted into the x-th input field, which would mean, you would need to call:

```ks
filterBar.addFilter('extension', 'is one of', ['jpg', 'png']); // for entering multiple values in the same field
filterBar.addFilter('bytes', 'is between', '500', '1000'); // for entering multiple values in different fields
```